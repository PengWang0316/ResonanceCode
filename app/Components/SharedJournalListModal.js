import React from 'react';

import JournalRow from './JournalRow';

const SharedJournalListModal = ({ reading }) => (
  <div className="modal fade" id="sharedJournalListModal" tabIndex="-1" role="dialog" aria-labelledby="sharedJournalListModalLable" aria-hidden="true">
    <div className="modal-dialog modal-lg" role="document">
      <div className="modal-content">
        <div className="modal-header">
          <h5 className="modal-title" id="sharedJournalListModalLable">{reading && <span>{`${reading.reading_name}'s shared journals'`}</span>}</h5>
          <button type="button" className="close" data-dismiss="modal" aria-label="Close">
            <span aria-hidden="true">&times;</span>
          </button>
        </div>
        <div className="modal-body">
          {reading && reading.journal_entries.map(journal =>
            <JournalRow key={journal._id} journal={journal} isSharedJournal />)}
        </div>
        <div className="modal-footer">
          <button type="button" className="btn btn-secondary" data-dismiss="modal">Close</button>
        </div>
      </div>
    </div>
  </div>
);
export default SharedJournalListModal;
