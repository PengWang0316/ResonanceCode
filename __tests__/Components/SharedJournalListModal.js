import React from 'react';
import renderer from 'react-test-renderer';
// import { shallow } from 'enzyme';

import SharedJournalListModal from '../../app/Components/SharedJournalListModal';

jest.mock('../../app/Components/JournalRow', () => 'JournalRow');

describe('SharedJournalListModal test', () => {
  const defaultProps = {
    reading: { reading_name: 'readingName', journal_entries: [{ _id: 'idA' }, { _id: 'idB' }] }
  };
  // const getShallowComponent = (props = defaultProps) => shallow(<SharedJournalListModal {...props} />);

  test('SharedJournalListModal snapshot', () => expect(renderer.create(<SharedJournalListModal {...defaultProps} />).toJSON()).toMatchSnapshot());
});
