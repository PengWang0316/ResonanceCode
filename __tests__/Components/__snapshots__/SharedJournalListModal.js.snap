// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SharedJournalListModal test SharedJournalListModal snapshot 1`] = `
<div
  aria-hidden="true"
  aria-labelledby="sharedJournalListModalLable"
  className="modal fade"
  id="sharedJournalListModal"
  role="dialog"
  tabIndex="-1"
>
  <div
    className="modal-dialog modal-lg"
    role="document"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header"
      >
        <h5
          className="modal-title"
          id="sharedJournalListModalLable"
        >
          <span>
            readingName's shared journals'
          </span>
        </h5>
        <button
          aria-label="Close"
          className="close"
          data-dismiss="modal"
          type="button"
        >
          <span
            aria-hidden="true"
          >
            Ã—
          </span>
        </button>
      </div>
      <div
        className="modal-body"
      >
        <JournalRow
          isSharedJournal={true}
          journal={
            Object {
              "_id": "idA",
            }
          }
        />
        <JournalRow
          isSharedJournal={true}
          journal={
            Object {
              "_id": "idB",
            }
          }
        />
      </div>
      <div
        className="modal-footer"
      >
        <button
          className="btn btn-secondary"
          data-dismiss="modal"
          type="button"
        >
          Close
        </button>
      </div>
    </div>
  </div>
</div>
`;
